src/common/NetworkManager.ts

```diff
@@ -242,8 +246,14 @@ export class NetworkManager extends EventEmitter {
       if (interceptionId) {
         this._onRequest(event, interceptionId);
         this._requestIdToInterceptionId.delete(requestId);
-      } else {
+      }
+      if (
+        !interceptionId ||
+        (this._requestIdToRequestPausedEvent.has(requestId) &&
+          event.type === 'Font')
+      ) {
         this._requestIdToRequestWillBeSentEvent.set(event.requestId, event);
+        this._requestIdToRequestPausedEvent.delete(requestId);
```

Comment:
Is it possible that this map will grow indefinitely? E.g., is _onRequestWillBeSent called if a request is aborted?

Reply:
See https://github.com/puppeteer/puppeteer/pull/6996#issuecomment-813797393

Variant A (stylesheets etc.)
- `_onRequestWillBeSent`
- `_onRequestPaused`

Variant B (fonts, cache disabled)
- `_onRequestPaused`
- `_onRequestWillBeSent`

Variant C (fonts, cache enabled)
- `_onRequestPaused`
- `_onRequestWillBeSent`
- `_onRequestPaused` <-- addressed by this PR

The branch of `_onRequestPaused` in question only runs if it was called before `_onRequestWillBeSent`, so only for variant B and the first call in variant C.
The subsequent call to `_onRequestWillBeSent` will consume `_requestIdToRequestPausedEvent`.
It will also consume `_requestIdToInterceptionId` but only if `event.type === 'Font'`.
See https://github.com/puppeteer/puppeteer/issues/7038#issuecomment-814069760 for context on why this additional condition was added. Maybe there is a better solution but for now it solves the issue.

Reply:
We could remove the entry in `_onLoadingFinished` and `_onLoadingFailed`, one of them should be called in any case.